File: mo-auth/src/main/java/ink/ckx/mo/auth/handler/MyAccessDeniedHandler.java
package ink.ckx.mo.auth.handler;

import ink.ckx.mo.auth.util.OAuth2EndpointUtils;
import ink.ckx.mo.common.core.result.ResultCode;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.security.access.AccessDeniedException;
import org.springframework.security.web.access.AccessDeniedHandler;

import java.io.IOException;

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/06
 */
public class MyAccessDeniedHandler implements AccessDeniedHandler {

    @Override
    public void handle(HttpServletRequest request, HttpServletResponse response, AccessDeniedException accessDeniedException) throws IOException, ServletException {
        OAuth2EndpointUtils.fail(response, ResultCode.ACCESS_DENIED);
    }
}

File: mo-auth/src/main/kotlin/ink/ckx/mo/auth/handler/MyAccessDeniedHandler.kt
package ink.ckx.mo.auth.handler

import ink.ckx.mo.auth.util.OAuth2EndpointUtils.fail
import ink.ckx.mo.common.core.result.ResultCode
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.access.AccessDeniedException
import org.springframework.security.web.access.AccessDeniedHandler

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/06
 */
class MyAccessDeniedHandler : AccessDeniedHandler {

    override fun handle(
        request: HttpServletRequest,
        response: HttpServletResponse,
        accessDeniedException: AccessDeniedException
    ) {
        fail(response, ResultCode.ACCESS_DENIED)
    }
}